model/DataQuery.php

```diff
@@ -661,6 +664,15 @@ public function applyRelation($relation) {
 					$this->query->addLeftJoin($componentBaseClass,
 						"\"$relationTable\".\"$componentField\" = \"$componentBaseClass\".\"ID\"");
 				}
+				else {
+					$alias = uniqid($componentBaseClass);
+					$this->query->addLeftJoin(
+						$componentBaseClass,
+						"\"$relationTable\".\"$componentField\" = \"$alias\".\"ID\"",
+						$alias
+					);
+					$this->lastAlias = $alias;
+				}
```

Comment:
This code to use an alias is in the `else` block, so what we’re saying is if we’re already joined to this table, still perform the join again buy instead use an alias.

I wondered earlier if we could remove the `if (isjoinedto)` statement above this and instead check whether there’s going to be a “duplicate” join, then rename the fields appropriately? I’ve not checked if we have enough info to do that, just thinking aloud :)